articles/ssl-tls/index.md

```diff
@@ -0,0 +1,75 @@
+Two of the most widely used encryption techniques are briefly explained below:
+
+    - **Symmetric key encryption** is the technique of encrypting information using cryptographic algorithms that use the same key for encryption and decryption. Such algorithms are called symmetric ciphers. When Alice and Bob communicate, they encrypt and decrypt their messages using a shared key. Care should be taken that this shared key isn’t compromised because it allows anybody to decrypt the ciphertext.
+
+        ![Symmetric key encryption](/engineering-education/ssl-tls/Sym.png)
+
+    - **Asymmetric key encryption** (also called public-key encryption) is the technique of encrypting information using cryptographic algorithms that use different keys for encryption and decryption. Such algorithms are called asymmetric ciphers. When Alice sends a message to Bob, she encrypts it using Bob’s public key. Bob decrypts Alice’s message using his private key, which isn’t known to anybody but him. Asymmetric encryption is more secure than symmetric encryption. However, it is computationally more expensive.
+
+         ![Asymmetric key encryption](/engineering-education/ssl-tls/Asym.png)
+
+2. **Cipher Suites** are a set of algorithms that are used to secure information being exchanged by the end points. A cipher suite announces the key exchange algorithm, the authentication algorithm, the bulk encryption algorithm and the Message Authentication Code (MAC) algorithm it supports. For example, the  `ECDHE-ECDSA-AES256-GCM-SHA384` suite implies that ECDHE is the key exchange algorithm, ECDSA is the authentication algorithm, AES256-GCM is the bulk encryption algorithm and SHA384 is the MAC algorithm it uses. Cipher suites are updated with safer algorithms from time to time. You can find out more about cipher suites [here](https://www.venafi.com/blog/what-are-cipher-suites).
+
+3. **SSL certificates** are digital data files that bind an organisation/domain name to a cryptographic key. SSL certificates are issued by Certificate Authorities (CAs) after verification of the data provided by the organisation. By signing and issuing an SSL certificate, the CA assures that the information enclosed in it is valid. 
+
+**How does SSL/TLS work?**
+
+As discussed before, public-key encryption requires high compute and adds considerable overhead. SSL/TLS, thus, makes use of public-key encryption just in the beginning, to exchange session keys. These session keys are symmetric in nature and are used to carry out the rest of the communication. SSL/TLS handshake is a part of an SSL/TLS exchange wherein the two end points exchange messages and establish the details of their communication. Since SSL/TLS is most widely used to encrypt web traffic, we’ll use a web client and server as examples of end points.
```

Comment:
```suggestion
As discussed before, public-key encryption requires high compute and adds considerable overhead. SSL/TLS, thus, makes use of public-key encryption just in the beginning, to exchange session keys. These session keys are symmetric in nature and are used to carry out the rest of the communication. SSL/TLS handshake is a part of an SSL/TLS exchange wherein the two endpoints exchange messages and establishes the details of their communication. Since SSL/TLS is most widely used to encrypt web traffic, we’ll use a web client and server as examples of endpoints.
```